### PR TITLE
Add update on dynamic import()

### PR DESCRIPTION
**Pull request by mathiasbynens**

### CODE REVIEW

**Comment by mathiasbynens** [comment]
All those `<script>` tags occur within code examples.

**Comment by mathiasbynens** [comment]
All those `<script>` tags occur within code examples.

File: src/content/en/updates/2017/11/dynamic-import.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
+project_path: /web/_project.yaml
+book_path: /web/updates/_book.yaml
+description: Dynamic import() introduces a new function-like form of import that unlocks new capabilities compared to static import. This article compares the two and gives an overview of what’s new.
```
Comment (ebidel) [inline src/content/en/updates/2017/11/dynamic-import.md]: kill the smart quotes just to be safe

File: src/content/en/updates/2017/11/dynamic-import.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+};
+
+// Named export `doStuff`
+export const doStuff = () => {
+  console.log('Doing stuff…');
+};
+```
+
+Here’s how to statically import and use the `./utils.js` module:
+
```
Comment (ebidel) [inline src/content/en/updates/2017/11/dynamic-import.md]: smart quote

File: src/content/en/updates/2017/11/dynamic-import.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+In some cases, it’s useful to:
+
+* import a module on-demand (or conditionally)
+* compute the module specifier at runtime
+* import a module from within a regular script (as opposed to a module)
+
+None of those are possible with static `import`.
+
+## Dynamic `import()` 🔥
```
Comment (ebidel) [inline src/content/en/updates/2017/11/dynamic-import.md]: `## Dynamic `import()` 🔥 {: #dynamic }` so this links properly.

File: src/content/en/updates/2017/11/dynamic-import.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+[Dynamic `import()`](https://github.com/tc39/proposal-dynamic-import) introduces a new
+function-like form of `import` that caters to those use cases. `import(moduleSpecifier)` returns a
+promise for the module namespace object of the requested module, which is created after fetching,
```
Comment (ebidel) [inline src/content/en/updates/2017/11/dynamic-import.md]: promise -> Promise

File: src/content/en/updates/2017/11/dynamic-import.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+instantiating, and evaluating all of the module’s dependencies, as well as the module itself.
+
+Here’s how to dynamically import and use the `./utils.js` module:
```
Comment (ebidel) [inline src/content/en/updates/2017/11/dynamic-import.md]: so smart! Maybe it's not an issue. cc @petele

Comment (mathiasbynens) [reply src/content/en/updates/2017/11/dynamic-import.md]: All my previous articles have had smart quotes. It’s not an issue.

File: src/content/en/updates/2017/11/dynamic-import.md (hunk 3 above)
Comment (mathiasbynens) [reply src/content/en/updates/2017/11/dynamic-import.md]: Ack

Comment (mathiasbynens) [reply src/content/en/updates/2017/11/dynamic-import.md]: Added `{: #static }` to the other header. Thanks!

File: src/content/en/updates/2017/11/dynamic-import.md (hunk 4 above)
Comment (mathiasbynens) [reply src/content/en/updates/2017/11/dynamic-import.md]: Lowercase is consistent with https://developers.google.com/web/fundamentals/primers/async-functionsand https://developers.google.com/web/fundamentals/primers/promises and with what you’d say for e.g. strings (“return a string”, not “return a `String`”)

File: src/content/en/updates/2017/11/dynamic-import.md (hunk 1 above)
Comment (mathiasbynens) [reply src/content/en/updates/2017/11/dynamic-import.md]: They’re not a problem

File: src/content/en/updates/2017/11/dynamic-import.md (hunk 2 above)
Comment (mathiasbynens) [reply src/content/en/updates/2017/11/dynamic-import.md]: They’re not a problem

**Comment by mathiasbynens** [comment]
Let me summarize all the comments so far to make it easier for new reviewers to step in:

* Re: @WebFundBot All those `<script>` tags occur within code examples.

* Re: @ebidel’s worry about smart quotes: my earlier posts all use them (as well as other non-ASCII Unicode symbols). The system handles it just fine. No need to avoid them :)

Re: “promise” vs. “`Promise`” — we had this discussion internally with @ebidel and @jakearchibald. Lowercase is consistent with https://developers.google.com/web/fundamentals/primers/async-functions and https://developers.google.com/web/fundamentals/primers/promises and with what you’d say for e.g. strings (“return a string”, not “return a `String`”). On IM last night, @ebidel said “it’s fine if we're already lowercasing [in] other places”.

With that, I think all feedback has been addressed.

File: src/content/en/updates/2017/11/dynamic-import.md (hunk 1 above)
Comment (PaulKinlan) [reply src/content/en/updates/2017/11/dynamic-import.md]: We don't use smart quotes

File: src/content/en/updates/2017/11/dynamic-import.md (hunk 2 above)
Comment (PaulKinlan) [reply src/content/en/updates/2017/11/dynamic-import.md]: We don't use smart quotes

File: src/content/en/updates/2017/11/dynamic-import.md (hunk 5 above)
Comment (PaulKinlan) [reply src/content/en/updates/2017/11/dynamic-import.md]: We don't use smart quotes. They shouldn't have been in the other posts either.

Comment (mathiasbynens) [reply src/content/en/updates/2017/11/dynamic-import.md]: Ok, I’ll change it. Just to satisfy my curiosity, what’s the rationale behind this? Is the goal to ban all non-ASCII characters from source? If it’s a strong requirement I could add a lint rule to `npm test`.

**Pull request merged**